dev/build-tool-support/src/main/scala/com/lightbend/lagom/dev/Reloader.scala

```diff
@@ -159,26 +160,46 @@ class Reloader(
   // Whether any source files have changed since the last request.
   @volatile private var changed = false
   // The last successful compile results. Used for rendering nice errors.
-  @volatile private var currentSourceMap = Option.empty[SourceMap]
+  @volatile private var currentSourceMap = Option.empty[Map[String, Source]]
   // A watch state for the classpath. Used to determine whether anything on the classpath has changed as a result
   // of compilation, and therefore a new classloader is needed and the app needs to be reloaded.
   @volatile private var watchState: WatchState = WatchState.empty
 
+  // Stores the most recent time that a file was changed
+  private val fileLastChanged = new AtomicReference[Instant]()
+
   // Create the watcher, updates the changed boolean when a file has changed.
   private val watcher = fileWatchService.watch(monitoredFiles, () => {
+
     changed = true
     onChange()
   })
   private val classLoaderVersion = new java.util.concurrent.atomic.AtomicInteger(0)
 
-  private val listeners = new java.util.concurrent.CopyOnWriteArrayList[Unit => Unit]()
+  private val quietTimeTimer = new Timer("reloader-timer", true)
+
+  private val listeners = new java.util.concurrent.CopyOnWriteArrayList[() => Unit]()
 
+  private val quietPeriodMs = 200l
   private def onChange(): Unit = {
-    import scala.collection.JavaConverters._
-    listeners.iterator().asScala.foreach(listener => listener())
+    val now = Instant.now()
+    fileLastChanged.set(now)
```

Comment:
> the expression fileLastChanged.compareAndSet(start, null) will return false, indicating that there was no change during the quiet period, which is wrong.

That's exactly the point, and it's exactly what's supposed to happen, returning false doesn't mean there were no changes, it means there *were* changes, which tells it that it wasn't quiet during the quiet period, which means there was no quiet period, which means it shouldn't execute.  What I'm trying to prevent here is a situation where the IDE saves 20 files in a short time, that triggers 20 file changed notifications, which causes the app to recompile 20 times.  Because sbt has good incremental compilation support, we never even noticed that this was happening before, but with maven, it's really slow and causes a big problem.  So what we want to do is ensure that only the *last* file changed notification in batch of notifications causes a recompile, and we do that using this quite period timeout mechanism.